components/carousel/index.jsx

```diff
@@ -0,0 +1,287 @@
+		this.setTranslationAmount(
+			-(this.stageWidth * (this.state.currentPanel - 1))
+		);
+	};
+
+	canGoToNext = () => this.state.currentPanel < this.nrOfPanels;
+
+	canGoToPrevious = () => this.state.currentPanel > 1;
+
+	handleKeyDown = (event) => {
+		const keyDownCallbacks = {
+			[KEYS.LEFT]: () => {
+				if (this.canGoToPrevious()) {
+					this.onPreviousPanelHandler();
+				}
+			},
+			[KEYS.RIGHT]: () => {
+				if (this.canGoToNext()) {
+					this.onNextPanelHandler();
+				}
+			},
+		};
+
+		if (keyDownCallbacks[event.keyCode]) {
+			EventUtil.trapImmediate(event);
+			keyDownCallbacks[event.keyCode](event.shiftKey ? 10 : 1);
+		}
+	};
+	render() {
+		const {
+			hasAutoplay,
+			hasPreviousNextPanelNavigation,
+			isInfinite,
+		} = this.props;
+		const id = this.props.id || this.generatedId;
+		const isPreviousBtnDisabled = !(isInfinite || this.canGoToPrevious());
+		const isNextBtnDisabled = !(isInfinite || this.canGoToNext());
+		const itemWidth = this.stageWidth / this.props.itemsPerPanel;
+
+		return (
+			<div className="slds-carousel" id={id} onKeyDown={this.handleKeyDown}>
+				<div className="slds-grid_vertical slds-col slds-path__scroller">
+					{hasAutoplay && (
+						<AutoPlayButton
+							isAutoPlayOn={this.state.isAutoPlayOn}
+							onClick={this.onAutoPlayBtnClick}
+						/>
+					)}
+					<div className="slds-grid slds-grid_align-center">
+						{hasPreviousNextPanelNavigation && (
+							<PreviousNextCarouselNavigator
+								icon="/assets/icons/utility-sprite/svg/symbols.svg#left"
+								assistiveText={this.props.assistiveText.previousPanel}
+								isDisabled={isPreviousBtnDisabled}
+								onClick={this.onPreviousPanelHandler}
+							/>
+						)}
+						<div
+							ref={this.stageItem}
+							className="slds-carousel__stage slds-align_absolute-center"
+						>
+							<div
+								className="slds-carousel__panels slds-is-relative"
+								style={{
+									transform: `translateX(${this.state.translateX}px)`,
+								}}
+							>
+								{this.props.items.map((item) => (
+									<CarouselItem
+										onRenderItem={this.props.onRenderItem}
+										{...item}
+										itemWidth={itemWidth}
+										key={item.id}
+									/>
+								))}
+							</div>
+						</div>
+						{hasPreviousNextPanelNavigation && (
+							<PreviousNextCarouselNavigator
+								icon="/assets/icons/utility-sprite/svg/symbols.svg#right"
+								assistiveText={this.props.assistiveText.nextPanel}
+								isDisabled={isNextBtnDisabled}
+								onClick={this.onNextPanelHandler}
+							/>
+						)}
+					</div>
+					<CarouselIndicators
+						style={this.props.indicatorStyles}
+						noOfIndicators={this.nrOfPanels}
+						currentIndex={this.state.currentPanel}
+						onClick={this.onIndicatorClickHandler}
+					/>
+				</div>
+			</div>
+		);
+	}
+}
+
+Carousel.displayName = CAROUSEL;
+
+Carousel.propTypes = {
+	/**
+	 * Array of objects with shape, needed for building a carousel item.
```

Comment:
Please alphabetize props to aid in finding one.

Reply:
done